--- c_src.orig/deps/js/src/config/SunOS5.10.mk	2010-02-10 14:11:41.000000000 -0800
+++ c_src/deps/js/src/config/SunOS5.10.mk	2010-02-10 14:13:34.000000000 -0800
@@ -47,4 +47,10 @@
 OTHER_LIBS += -L/usr/sfw/lib/mozilla -R/usr/sfw/lib/mozilla
 
 CC=gcc
+LD         = gcc
+MKSHLIB    = $(LD) -shared
+ASFLAGS   += -xarch=v9 -DULTRA_SPARC
+DEFINES   += -DULTRA_SPARC
+OS_CFLAGS += -fPIC -Wa,-xarch=v9,-DULTRA_SPARC
+CFLAGS    += -DULTRA_SPARC -fPIC

