--- c_src.orig/js/src/config.mk	2008-03-12 10:36:06.000000000 -0400
+++ c_src/js/src/config.mk	2011-03-30 20:11:30.000000000 -0400
@@ -100,10 +100,26 @@
 ifeq ($(OS_ARCH),Darwin)
 OS_CONFIG      := Darwin
 else
+ifeq ($(OS_ARCH), NetBSD)
+OS_CONFIG      := NetBSD
+else
+ifeq ($(OS_ARCH), FreeBSD)
+OS_CONFIG      := FreeBSD
+else
+ifeq ($(OS_ARCH), DragonFly)
+OS_CONFIG      := DragonFly
+else
+ifeq ($(OS_ARCH), OpenBSD)
+OS_CONFIG      := OpenBSD
+else
 OS_CONFIG       := $(OS_ARCH)$(OS_OBJTYPE)$(OS_RELEASE)
 endif
 endif
 endif
+endif
+endif
+endif
+endif
 
 ASFLAGS         =
 DEFINES         =
@@ -141,7 +157,9 @@
 
 SO_SUFFIX = so
 
+ifneq ($(OS_ARCH),SunOS)
 NS_USE_NATIVE = 1
+endif
 
 # Java stuff
 CLASSDIR     = $(DEPTH)/liveconnect/classes
