--- c_src/nsprpub.old/configure.in	2011-03-31 21:51:51 -0400
+++ c_src/nsprpub/configure.in	2011-03-31 21:53:18 -0400
@@ -1161,6 +1161,29 @@
     MDCPUCFG_H=_freebsd.cfg
     PR_MD_CSRCS=freebsd.c
     ;;
+*-dragonfly*)
+		if test -z "$USE_NSPR_THREADS"; then
+			USE_PTHREADS=1
+		fi
+		AC_DEFINE(XP_UNIX)
+		AC_DEFINE(FREEBSD)
+		AC_DEFINE(HAVE_BSD_FLOCK)
+		AC_DEFINE(HAVE_SOCKLEN_T)
+		CFLAGS="$CFLAGS $(DSO_CFLAGS) -ansi -Wall"    MOZ_OBJFORMAT=`test -x /usr/bin/objformat && /usr/bin/objformat || echo aout`
+		if test "$MOZ_OBJFORMAT" = "elf"; then
+			DLL_SUFFIX=so
+		else
+			DLL_SUFFIX=so.1.0
+		fi
+		MKSHLIB='$(CC) $(DSO_LDOPTS) -o $@'
+		DSO_CFLAGS=-fPIC
+		DSO_LDOPTS='-shared -Wl,-soname -Wl,$(notdir $@)'
+		MDCPUCFG_H=_freebsd.cfg
+		PR_MD_CSRCS=freebsd.c
+		if test "$LIBRUNPATH"; then
+			DSO_LDOPTS="$DSO_LDOPTS -Wl,-R$LIBRUNPATH"
+		fi
+		;;
 
 *-hpux*)
     AC_DEFINE(XP_UNIX)
@@ -2511,7 +2534,7 @@
 	if test -z "`egrep -i '(unrecognize|unknown)' conftest.out | grep pthread`" && test -z "`egrep -i '(error|incorrect)' conftest.out`" ; then
 	    ac_cv_have_dash_pthread=yes
 		case "$target_os" in
-	    freebsd*)
+	    freebsd* | dragonfly*)
 # Freebsd doesn't use -pthread for compiles, it uses them for linking
             ;;
 	    *)
@@ -2549,7 +2572,7 @@
             _PTHREAD_LDFLAGS=
         fi
 	    ;;
-    *-freebsd*)
+    *-freebsd* | dragonfly*)
 	    AC_DEFINE(_REENTRANT)
 	    AC_DEFINE(_THREAD_SAFE)
 	    dnl -pthread links in -lc_r, so don't specify it explicitly.
@@ -2630,7 +2653,7 @@
         AC_DEFINE(_PR_NEED_PTHREAD_INIT)
     fi
     ;;
-*-freebsd*)
+*-freebsd* | dragonfly*)
     if test -n "$USE_NSPR_THREADS"; then
         AC_DEFINE(_PR_LOCAL_THREADS_ONLY)
     fi
